### ISSUE TITLE
2D movement project can be replaced with a demo

### ISSUE THREAD

**Issue by YuriSizov** [godotengine/godot-docs-project-starters#2]
It seems that the 2D movement project is used in the article more like a demo than like a starter template. So it should probably be removed and replaced with a demo project of a similar kind.

https://github.com/godotengine/godot-docs-project-starters/tree/master/src/2d_movement_starter
https://docs.godotengine.org/en/latest/tutorials/2d/2d_movement.html

**Comment by jynus** [comment]
Hi, @YuriSizov  I would like to help here, but I am not sure I understand the overall philosophy of what's being proposed (beyond just this specific file).

As I understand, the idea is to create here just skeleton projects that readers can use as a starting point and code along the manual/tutorials, is that right?

However, I see a problem, and that is that some people will want to download a final version of those "code along" exercises (because some people will be able to see how it is exactly done, or if they have put a file in the wrong location, etc.

I see 3 options for that:

* Focus on a skeleton/starter project (I believe this is what is being proposed)
* Create 2 downloable (or more!) files, a skeleton and a finished project (or potentially, some intermediate steps)
* Provide the skeleton and build into a demo, that people can download in the demos repo (and we can link to it?)

My guess is that the "demo" was supposed to be used as both a skeleton and a final code by just adding a new scene (but that would need some better explanation)- that could be a 4th approach.

Let me know your thoughts/clear my misunderstandings so maybe I can help with this.

There is a demo accessibility issue, though- github is not a friendly site for those wanting to just download a .zip, and the asset library still offers the 3.X demos (but that could be a separate issue).

**Comment by Calinou** [comment]
> and the asset library still offers the 3.X demos (but that could be a separate issue).

See https://github.com/godotengine/godot-demo-projects/issues/920.

**Comment by YuriSizov** [comment]
> As I understand, the idea is to create here just skeleton projects that readers can use as a starting point and code along the manual/tutorials, is that right?

Yes, projects provided in tutorials should serve tutorials they are included in. Complete example projects can still be provided at the end of the article, but must be uploaded as a part of the demo projects repo instead, as that's what they are.

I don't have a prescribed solution here, maybe the article is fine as is and doesn't need a skeleton/starter project. But the point of this issue is that there is no reason to keep a complete project in the docs. If it's complete, then it should be put into demos, or perhaps there already is a demo and this project here is just duplicating it.

**Comment by jynus** [comment]
Thanks for the comments, now that I understood the intention and the core problem, it is ok to not have a definitive answer, I can try to come with a more or less consistent but simple approach that at least makes things better for both the readers -but also results on easier to review updates.

Thank you again!
